solutions/usaco-988.mdx

```diff
@@ -112,4 +115,65 @@ print(ans, file=open('photo.out', 'w'))
 ```
 
 </PySection>
+<JavaSection>
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+	
+```java
+import java.io.*;
+import java.util.*;
+
+public class Photoshoot {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("photo");
+
+		int N = io.nextInt();
+		int[] bessieList = new int[N];
+		int[] correctList = new int[N];
+		// Used to prevent repeated values in our permutation, which isn't right
+		List<Integer> isItThere = new ArrayList<>();
```

Comment:
a hashset would be better imo

Reply:
I think a list makes more sense conceptually, since a hashset is really only useful in this situation if you care about the number of times a given key appears, and we don't. But if you want me to use a hashset instead, that's cool.

Reply:
> since a hashset is really only useful in this situation if you care about the number of times a given key appears

i thought a hashset didn't allow duplicates?
and usually if you only want a collection for memberchecking, then a hashset would be ideal

Reply:
> i thought a hashset didn't allow duplicates?

oop you're right
> and usually if you only want a collection for memberchecking, then a hashset would be ideal

Why's that? (Just curious, since to my knowledge both hashsets and lists are O(1)

Reply:
wait since when did lists have an o(1) membership check

Reply:
Well that settles it

Reply:
uh im pretty sure arraylists have o(n) membership check cause they are internally stored as a static array

Reply:
so hashset is definitely more optimal

Reply:
nvm looks like you committed the change